pkg/netconfig/netconfig.go

```diff
@@ -0,0 +1,339 @@
+		if !newSet[s] {
+			removed = append(removed, s)
+		}
+	}
+
+	return
+}
+
+func InitDummyDevice() error {
+	out, _ := DefaultNetUtilsCommandExecutor.ExecuteIPRouteCommand("4", "link", "show")
+	if !strings.Contains(out.String(), " dummy0: ") {
+		_, err := DefaultNetUtilsCommandExecutor.ExecuteIPRouteCommand("4", "link", "add", "dummy0", "type", "dummy")
+		if err != nil {
+			return fmt.Errorf("Error creating dummy device: %v", err)
+		}
+		_, err = DefaultNetUtilsCommandExecutor.ExecuteIPRouteCommand("4", "link", "set", "dummy0", "up")
+		if err != nil {
+			return fmt.Errorf("Error setting up dummy device: %v", err)
+		}
+	}
+	if err := DefaultNetUtilsCommandExecutor.ExecuteIPTablesCommand("4", "-t", "mangle", "-C", "POSTROUTING", "-o", "dummy0", "-j", "LOGGING"); err != nil {
+		err = DefaultNetUtilsCommandExecutor.ExecuteIPTablesCommand("4", "-t", "mangle", "-A", "POSTROUTING", "-o", "dummy0", "-j", "LOGGING")
+		if err != nil {
+			return errors.New(fmt.Sprintf("Error creating ip%stables rule for logging packets to dummy device: %v\n", "", err))
+		}
+	}
+
+	if err := DefaultNetUtilsCommandExecutor.ExecuteIPTablesCommand("6", "-t", "mangle", "-C", "POSTROUTING", "-o", "dummy0", "-j", "LOGGING"); err != nil {
+		err = DefaultNetUtilsCommandExecutor.ExecuteIPTablesCommand("6", "-t", "mangle", "-A", "POSTROUTING", "-o", "dummy0", "-j", "LOGGING")
+		if err != nil {
+			return errors.New(fmt.Sprintf("Error creating ip%stables rule for logging packets to dummy device: %v\n", "6", err))
+		}
+	}
+	return nil
+}
+
+func GetBlackholeRoutes(ipVersion string) ([]string, error) {
+	blackholeRoutes := []string{}
+
+	ipOut, err := DefaultNetUtilsCommandExecutor.ExecuteIPRouteCommand(ipVersion, "route")
+
+	if err != nil {
+		return blackholeRoutes, err
+	}
+
+	lines := strings.Split(ipOut.String(), "\n")
+	for _, line := range lines {
+		if strings.Contains(line, "dummy0") && !strings.Contains(line, "fe80::/64") {
+			fields := strings.Fields(line)
+			addr := fields[0]
+			blackholeRoutes = append(blackholeRoutes, addr)
+		}
+	}
+
+	return blackholeRoutes, nil
+}
+
+func DeleteRoutes(ipVersion string, addrs []string) error {
+	errorOccurred := false
+	for _, addr := range addrs {
+		fmt.Printf("Delete blackhole route %s\n", addr)
+		out, err := DefaultNetUtilsCommandExecutor.ExecuteIPRouteCommand(ipVersion, "route", "del", addr)
+		if err != nil {
+			errorOccurred = true
+			fmt.Printf("error deleting blackhole route %s: %s %v\n", addr, out, err)
+		}
+	}
+	if errorOccurred {
+		return fmt.Errorf("Error deleting blackhole routes\n")
+	}
+	return nil
+}
+
+func AddRoutes(ipVersion string, addrs []string) error {
+	errorOccurred := false
+	for _, addr := range addrs {
+		fmt.Printf("Add blackhole route %s\n", addr)
+		out, err := DefaultNetUtilsCommandExecutor.ExecuteIPRouteCommand(ipVersion, "route", "add", addr, "dev", "dummy0")
+		if err != nil {
+			errorOccurred = true
+			fmt.Printf("error adding blackhole route %s: %s %v\n", addr, out, err)
```

Comment:
Logging the actual route entries may not be desired.

Reply:
Removed actual addresses.